packages/sdk/src/node.ts

```diff
@@ -0,0 +1,105 @@
+export function createScriptNode(
+  url: string,
+  cb: (error?: Error, scriptContext?: any) => void,
+  attrs?: Record<string, any>,
+  createScriptHook?: (url: string) => any | void,
+) {
+  if (createScriptHook) {
+    const hookResult = createScriptHook(url);
+    if (hookResult && typeof hookResult === 'object' && 'url' in hookResult) {
+      url = hookResult.url;
+    }
+  }
+
+  let urlObj: URL;
+  try {
+    urlObj = new URL(url);
+  } catch (e) {
+    console.error('Error constructing URL:', e);
+    cb(new Error(`Invalid URL: ${e}`));
+    return;
+  }
+  if (!globalThis.fetch) {
+    //TODO: this shouldnt be coupled to webpack require
+    //@ts-ignore
+    globalThis.fetch = __non_webpack_require__(
```

Comment:
Can we pass the require as parameter ? Because i am worry about it will not work in rspack or other bundlers

Reply:
Where should it be passed from? 
I agree this is too coupled to webpack, but did not see immdiate location of where to pass down require